app/exporthtmldialog.cpp

```diff
@@ -32,8 +33,9 @@ ExportHtmlDialog::ExportHtmlDialog(const QString &fileName, QWidget *parent) :
 
     if (!fileName.isEmpty()) {
         QFileInfo info(fileName);
-        QString exportFileName = info.absoluteFilePath().replace(info.suffix(), "html");
-        ui->exportToLineEdit->setText(exportFileName);
+        QString absFilePath = info.absoluteFilePath();
+        QString exportFileName = absFilePath.mid(0, absFilePath.lastIndexOf('.')) + ".html";
```

Comment:
[used QFileInfo::baseName()](https://github.com/cloose/CuteMarkEd/pull/329/commits/a0a9b2cb617679abb76a0c102ed12f1a72ff100c) , maybe in better way than you suggest :)
tested, path is generated correctly